microsoft-edge/progressive-web-apps-chromium/how-to/digital-goods-api.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Provide in-app purchases with the Digital Goods API
+description: How to provide in-app purchases in a Progressive Web App by using the Digital Goods API and the Payment Request API.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.service: microsoft-edge
+ms.subservice: pwa
+ms.date: 11/25/2024
+---
+# Provide in-app purchases with the Digital Goods API
+
+If your Progressive Web App (PWA) is listed in the Microsoft Store, you can provide in-app products and subscriptions by using the Digital Goods API and the Payment Request API.
+
+
+<!-- ====================================================================== -->
+## Digital Goods API
+
+The Digital Goods API is an interface between your PWA app and Microsoft Store.  The Digital Goods API supports:
+* Querying the details of a digital item from the Microsoft Store backend, such as the item's name, description, and regional price.
+* Consuming or acknowledging purchases.
+* Checking the digital items that are currently owned by the user.
+* Checking the purchase history of the user.
+
+See:
+* [Digital Goods API For Microsoft Store PWA Explainer](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/PwaDigitalGoods/explainer.md)
+* [Digital Goods API: Draft Community Group Report](https://wicg.github.io/digital-goods/)
+
+
+<!-- ====================================================================== -->
+## Payment Request API
+
+The Payment Request API⁠⁠ handles the actual payment transaction when a purchase is made by user.  The Payment Request API uses the item details that the Digital Goods API provides, to make the in-app purchase by using whichever billing payment method the user has set up at the Microsoft Store.
+
+See [Payment Request API](https://www.w3.org/TR/payment-request/) at W3C.
+
+
+<!-- ====================================================================== -->
+## Checking whether the Digital Goods API is available
+
+To detect whether you've correctly enabled the API on your website, check for the `getDigitalGoodsService` method in the window object:
```

Comment:
This is phrased as if the developer needs to take action to enable the API, but we're not telling them what action that is. Is there an origin trial? Is there a flag to enable somewhere? Does this come with a pre-release version of Edge?

Reply:
now doc'd:
https://review.learn.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/digital-goods-api?branch=pr-en-us-3324#enable-the-digital-goods-api
@captainbrosset